src/dae/daeUtils.cpp

```diff
@@ -169,8 +171,13 @@ static string tmpDir = string(getenv("TMPDIR"));
 string cdom::getRandomFileName() {
     std::string randomSegment;
     // have to createa a buffer in order to make it multi-thread safe
+#ifdef NO_BOOST
     std::string tmpbuffer; tmpbuffer.resize(L_tmpnam*2+1);
     std::string tmp(tmpnam(&tmpbuffer[0]));
+#else
+    boost::uuids::uuid uuid = boost::uuids::random_generator()();
+    std::string tmp(boost::uuids::to_string(uuid));
+#endif
```

Comment:
I guess the reason I am hesitant to have it fixed this way is that any solution to this warning will not actually be a more secure method of generating random file names.  The reason `tmpnam()` is deprecated is because the whole idea of generating a temp filename separately from opening it is insecure.  this function cannot be fixed without deleting it, and I don't think we want to start undertaking the refactoring of colladadom that would be required to remove this function.  I think it would be better to just use appropriate `#pragmas` to suppress the warning

Reply:
like this:
https://github.com/secondlife/3p-colladadom/compare/brad/tmpnam-warnings